### PR TITLE
[Merged by Bors] - feat(Kernel): `map (κ ×ₖ η) Prod.swap = η ×ₖ κ`

### PR DESCRIPTION
**Pull request by YaelDillies**
and similar lemmas.

From PFR

Co-authored-by: Rémy Degenne

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Probability/Kernel/Composition.lean [hunk 1]
```diff
@@ -1192,6 +1222,28 @@ instance IsSFiniteKernel.prod (κ : Kernel α β) (η : Kernel α γ) :
     snd (κ ×ₖ η) = η := by
   ext x; simp [snd_apply, prod_apply]
 
+lemma comap_prod_swap (κ : Kernel α β) (η : Kernel γ δ) [IsFiniteKernel κ] [IsFiniteKernel η] :
```
Comment (sgouezel) [inline Mathlib/Probability/Kernel/Composition.lean:1225]:
```suggestion
lemma comap_prod_swap (κ : Kernel α β) (η : Kernel γ δ) [IsSFiniteKernel κ] [IsSFiniteKernel η] :
```

File: Mathlib/Probability/Kernel/Composition.lean [hunk 2]
```diff
@@ -238,10 +238,44 @@ lemma compProd_zero_right (κ : Kernel α β) (γ : Type*) {mγ : MeasurableSpac
...
+  simp_rw [this]
+  rw [lintegral_indicator_const hs, one_mul]
+
+lemma compProd_deterministic_apply [MeasurableSingletonClass γ] {f : α × β → γ} (hf : Measurable f)
+    {s : Set (β × γ)} (hs : MeasurableSet s) (κ : Kernel α β) [IsSFiniteKernel κ] (x : α) :
+    (κ ⊗ₖ deterministic f hf) x s = κ x {b | (b, f (x, b)) ∈ s} := by
+  rw [compProd_apply hs]
+  simp only [deterministic_apply, measurableSet_setOf, Set.mem_setOf_eq, Measure.dirac_apply]
+  classical
+  simp only [Set.mem_setOf_eq, Set.indicator_apply, Pi.one_apply]
```
Comment (sgouezel) [inline Mathlib/Probability/Kernel/Composition.lean:263]:
```suggestion
  simp only [deterministic_apply, measurableSet_setOf, Set.mem_setOf_eq, Measure.dirac_apply,
    Set.mem_setOf_eq, Set.indicator_apply, Pi.one_apply, compProd_apply hs]
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Probability/Kernel/Composition.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Probability/Kernel/Composition.lean:263]: Nicely golfed

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**